asdf/tests/test_compression.py

```diff
@@ -67,7 +70,7 @@ def _roundtrip(tmpdir, tree, compression=None,
     with asdf.open(generic_io.InputStream(buff), **read_options) as ff:
         helpers.assert_tree_match(tree, ff.tree)
 
-    return ff
+    return tmpfile
```

Comment:
Is there a reason this now returns the path to the file instead of the tree itself?  I see that does not affect the legacy tests as this `_roundtrip` function is essentially a wrapper for the `assert_tree_match` assertion helper.

Perhaps it shouldn't return anything and be formalized as an assertion helper itself?

Reply:
I think this version of the tests just uses the filename to open the file again to check that the compression extension shows up in the history (`test_compression_with_extension()`).  The old `return ff` was actually returning a closed ASDF object, so the history couldn't be queried.  I didn't want to return an open ASDF object; we could instead return the history, or nothing, and make the history check separate.

Reply:
Ah, I see now.  Thanks.  Yes, that's an improvement!  😄